app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala

```diff
@@ -155,8 +155,8 @@ object FormTemplateValidator {
     }
 
     formTemplate.authConfig match {
-      case HmrcSimpleModule | HmrcAgentModule(_) if !userContextComponentType.isEmpty =>
```

Comment:
This validation should be reversed ie. we should match valid cases, not the invalid ones. Current implementation is wrong wrt to eeitt auth.
It should be something like this:
```
if (userContextComponentType.nonEmpty) {
  formTemplate.authConfig match {
    case HmrcEnrolmentModule(_) | HmrcAgentWithEnrolmentModule(_, _) => Valid
    case _ => Invalid("You used '${user.enrolledIdentifier}' but you didn't provide 'serviceId'.")
  }
} else Valid
```This validation should be reversed ie. we should match valid cases, not the invalid ones. Current implementation is wrong wrt to eeitt auth.
It should be something like this:
```
if (userContextComponentType.nonEmpty) {
  formTemplate.authConfig match {
    case HmrcEnrolmentModule(_) | HmrcAgentWithEnrolmentModule(_, _) => Valid
    case _ => Invalid("You used '${user.enrolledIdentifier}' but you didn't provide 'serviceId'.")
  }
} else Valid
```

Reply:
What about 'Anonymous'?
should not be 
"case HmrcSimpleModule | HmrcAgentModule(_) | Anonymous => "

Reply:
Combination of `${user.enrolledIdentifier}` and `anonymous` is indeed invalid as per my snippet.

Reply:
I have changed the implementation + . I gave a good refactor to the test - however there is a test for Anonymous scenario missing. Anyway let me know